### PR TITLE
[PAY-881] Mobile messages (chats) list screen

### PR DESCRIPTION
**Pull request by dharit-tan**
### Description
First screen showing list of chats.
Simulator Screen Shot - iPhone 14 Pro - 2023-01-31 at 15 29 07

### Dragons

*Is there anything the reviewer should be on the lookout for? Are there any dangerous changes?*

### How Has This Been Tested?

ios simulator.

### How will this change be monitored?

*For features that are critical or could fail silently please describe the monitoring/alerting being added.*

### Feature Flags ###

*Are all new features properly feature flagged? Describe added feature flags.*

### CODE REVIEW

File: packages/mobile/src/screens/chat-screen/ChatListScreen.tsx [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+  useEffect(() => {
+    dispatch(chatActions.fetchMoreChats())
+  }, [dispatch])
+
+  return (
+    <Screen
+      url='/chat'
+      // variant='secondary'
+      // icon={IconChat}
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatListScreen.tsx:39]: nit: comments?

File: packages/mobile/src/screens/chat-screen/ChatListScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      title={messages.title}
+      style={styles.title}
+    >
+      <ScreenContent>
+        <ScrollView style={styles.root}>
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatListScreen.tsx:44]: There doesn't seem to be a root style

File: packages/mobile/src/screens/chat-screen/ChatListScreen.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+                  return (
+                    <ChatListItem
+                      key={chat.chat_id}
+                      currentChatId={chat.chat_id}
+                      chat={chat}
+                    />
+                  )
+                })
+              ) : (
+                <LoadingSpinner />
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatListScreen.tsx:58]: Should this be styled so it's centered?

File: packages/mobile/src/screens/chat-screen/ChatUser.tsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    flexDirection: 'row',
+    marginBottom: spacing(1)
+  },
+  userName: {
+    fontSize: typography.fontSize.large,
+    fontWeight: 'bold',
+    color: palette.neutral
+  },
+  userBadges: {
+    height: 22
```
Comment (rickyrombo) [inline packages/mobile/src/screens/chat-screen/ChatUser.tsx:38]: Should this be set explicitly? Also it looks like figma has these as 14x14?

**Review by rickyrombo — APPROVED** [review]

**Comment by rickyrombo** [comment]
Fix typecheck before merging!

File: packages/mobile/src/screens/chat-screen/ChatListScreen.tsx (hunk 3 above)
Comment (dharit-tan) [reply packages/mobile/src/screens/chat-screen/ChatListScreen.tsx:58]: Don't see mocks for this in the figma, was gonna ask design about this but I can at least center it for now.

File: packages/mobile/src/screens/chat-screen/ChatUser.tsx (hunk 4 above)
Comment (dharit-tan) [reply packages/mobile/src/screens/chat-screen/ChatUser.tsx:38]: Huh... I had that as a hacky solution to have them vertically centered.. but the issue seems to have gone away now. Removing this. And they default to 14x14.

**Pull request merged**